subsys/bluetooth/host/conn.c

```diff
@@ -903,40 +899,13 @@ void bt_conn_recv(struct bt_conn *conn, struct net_buf *buf, uint8_t flags)
 			bt_conn_reset_rx_state(conn);
 		}
 
-		if (!buf->len) {
-			BT_DBG("Empty ACL_START");
-			net_buf_unref(buf);
-			return;
-		}
-
-		if (buf->len < sizeof(len)) {
-			/* Got ACL data of size less than the length field.
-			 * Don't parse the length field until the next
-			 * BT_ACL_CONT when it might be valid (=>2 bytes).
-			 */
-			conn->rx = buf;
-			return;
-		}
-
-		hdr = (void *)buf->data;
-		len = sys_le16_to_cpu(hdr->len);
-
-		BT_DBG("First, len %u final %u", buf->len, len);
-
-		if (buf->len > (sizeof(*hdr) + len)) {
-			BT_ERR("L2CAP overflow");
-			bt_conn_reset_rx_state(conn);
-			net_buf_unref(buf);
-			return;
-		}
-
-		conn->rx_len = (sizeof(*hdr) + len) - buf->len;
-		BT_DBG("rx_len %u", conn->rx_len);
-		if (conn->rx_len) {
-			conn->rx = buf;
-			return;
-		}
+		BT_DBG("First, len %u final %u", buf->len,
+		       (buf->len < sizeof(uint16_t)) ?
+		       0 :
+		       sys_le16_to_cpu(
+			       ((struct bt_l2cap_hdr *)buf->data)->len));
```

Comment:
Actually it is just the length that should be read, not bt_l2cap_hdr, so this should probably be something like:

buf->len < sizeof(uint16_t) ? buf->len : sys_le16_to_cpu(buf->data);

Reply:
I originally did this, but changed it to make it clearer that it's the l2cap header length field we're retrieving. No problem switching it back though.